File: linked-list/linked_list.rb
# linked_list.rb
class Node
  attr_accessor :data, :next

  def initialize(data)
    @data = data
    @next = nil
  end
end

class LinkedList
  attr_accessor :head

  def initialize
    @head = nil
  end

  def append(data)
    new_node = Node.new(data)
    if @head.nil?
      @head = new_node
    else
      current = @head
      current = current.next while current.next
      current.next = new_node
    end
  end

  def print_list
    current = @head
    while current
      print "#{current.data} "
      current = current.next
    end
    puts ""
  end
end

llist = LinkedList.new
llist.append(1)
llist.append(2)
llist.append(3)
llist.append(4)
llist.print_list

File: linked-list/go-linked-list/linked_list.go
package main

import "fmt"

type Node struct {
    data int
    next *Node
}

type LinkedList struct {
    head *Node
}

func (ll *LinkedList) append(data int) {
    newNode := &Node{data: data}
    if ll.head == nil {
        ll.head = newNode
    } else {
        current := ll.head
        for current.next != nil {
            current = current.next
        }
        current.next = newNode
    }
}

func (ll *LinkedList) printList() {
    current := ll.head
    for current != nil {
        fmt.Print(current.data, " ")
        current = current.next
    }
    fmt.Println()
}

func main() {
    llist := LinkedList{}
    llist.append(1)
    llist.append(2)
    llist.append(3)
    llist.append(4)
    llist.printList()
}
